lib/src/computer.dart

```diff
@@ -17,29 +20,42 @@ class Computer {
   /// Returns `true` if `Computer` turned on and `false` otherwise
   bool get isRunning => _computeDelegate.isRunning;
 
-  /// Turn on `Computer`, `workersCount` should not be more than 0, default is 2
+  /// Turn on `Computer`, `workersCount` should be more than 0, default is 2
   /// `verbose` is false by default, enabling it leads to logging of every operation
   Future<void> turnOn({
     int workersCount = 2,
     bool verbose = false,
   }) async {
-    return _computeDelegate.turnOn(
+    assert(workersCount > 0, 'workersCount should be more than 0');
+    if (isRunning) return;
+    _completer = Completer<void>();
+    await _computeDelegate.turnOn(
       workersCount: workersCount,
       verbose: verbose,
     );
+    _completer?.complete();
   }
 
   /// Executes function `fn` with passed `param`. Takes only top-level functions and static methods.
   /// `P` is `param` type, `R` is function return type
+  /// Throws when turnOn has not been called or waits for turnOn completion
   Future<R> compute<P, R>(
     Function fn, {
     P? param,
   }) async {
+    if (_completer == null) {
+      throw StateError('Computer is not running, you must run turnOn first!');
+    }
+    await _completer!.future;
     return _computeDelegate.compute<P, R>(fn, param: param);
   }
 
   /// Turn off `Computer`
   Future<void> turnOff() async {
+    if (_completer?.isCompleted == false) {
+      _completer!.completeError('Computer turned off');
+    }
+    _completer = null;
```

Comment:
Should we wait for turn off to complete before marking this. and/or wait for turn on to complete if it was running at that time